src/connection/activate.test.ts

```diff
@@ -0,0 +1,256 @@
+import { Web3ReactHooks } from '@web3-react/core'
+import { Connector } from '@web3-react/types'
+
+import { act, renderHook } from '../test-utils/render'
+import { ActivationStatus, useActivationState } from './activate'
+import { Connection, ConnectionType } from './types'
+import { ErrorCode } from './utils'
+
+type DeferredPromise = {
+  promise: Promise<unknown>
+  resolve: () => void
+  reject: (reason: unknown) => void
+}
+function createDeferredPromise() {
+  const defferedPromise = {} as DeferredPromise
+
+  const promise = new Promise<void>((resolve, reject) => {
+    defferedPromise.reject = reject
+    defferedPromise.resolve = resolve
+  })
+  defferedPromise.promise = promise
+
+  return defferedPromise
+}
+
+class MockConnector extends Connector {
+  activate: () => void
+  deactivate: () => void
+  resetState = jest.fn()
+
+  constructor(activate: () => void, deactivate?: () => void) {
+    const actions = {
+      startActivation: jest.fn(),
+      update: jest.fn(),
+      resetState: jest.fn(),
+    }
+    super(actions)
+
+    this.activate = activate ?? jest.fn()
+    this.deactivate = deactivate ?? jest.fn()
+  }
+}
+
+function createMockConnection(
+  activate: () => void,
+  deactivate?: () => void,
+  type = ConnectionType.INJECTED
+): Connection {
+  return {
+    getName: () => 'Test Connection',
+    hooks: {} as unknown as Web3ReactHooks,
+    type,
+    shouldDisplay: () => true,
+    connector: new MockConnector(activate, deactivate),
+  }
+}
+
+beforeEach(() => {
+  console.error = jest.fn()
```

Comment:
You should update this to squelch anything you log so it doesn't spam the console. Right now you're using debug and log.
You should also include assertions for these in your tests if you want to make sure they're not removed.